### PR TITLE
bluetooth: hci: enable backends by default

### PR DESCRIPTION
**Pull request by JordanYates**
Enable backends by default, instead of requiring some other module to turn them on. This aligns with the behaviour of sensor drivers and `BT_RPMSG`.

Default backend for the `unit_testing` board is explicitly set to `BT_NO_DRIVER`, from the previously implicit `BT_RPMSG`.

### CODE REVIEW

**Review by Thalley — DISMISSED** [review]

File: subsys/testsuite/boards/unit_testing/unit_testing/Kconfig.defconfig [hunk 1]
```diff
@@ -2,4 +2,7 @@
 #
 # Copyright (c) 2022 Nordic Semiconductor
 
-# Intentionally left empty
+# Bluetooth unit tests expect this backend
+choice BT_HCI_BUS_TYPE
+	default BT_RPMSG
```
Comment (jori-nordic) [inline subsys/testsuite/boards/unit_testing/unit_testing/Kconfig.defconfig:7]: shouldn't that be `BT_NO_DRIVER`?

Comment (JordanYates) [reply subsys/testsuite/boards/unit_testing/unit_testing/Kconfig.defconfig:7]: Maybe conceptually, but the current default for `unit_testing` is `RPMSG`, and several tests depend on that symbols `select BT_HAS_HCI_VS`.

Comment (JordanYates) [reply subsys/testsuite/boards/unit_testing/unit_testing/Kconfig.defconfig]: Sure hope it doesn't open a can of worms

**Review by jori-nordic — APPROVED** [review]

File: subsys/testsuite/boards/unit_testing/unit_testing/Kconfig.defconfig (hunk 1 above)
Comment (jori-nordic) [reply subsys/testsuite/boards/unit_testing/unit_testing/Kconfig.defconfig]: Thanks! Ping me if it does.

**Review by Thalley — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**